mwdb/web/src/commons/ui/HexView.js

```diff
@@ -39,46 +39,42 @@ class HexViewNumberRenderer {
     }
 }
 
-export default class HexView extends Component {
-    constructor(props) {
-        super(props);
-        this.numberRenderer = new HexViewNumberRenderer();
-        this.editor = React.createRef();
-        this.originalContent = null;
-        this.hexlifiedContent = null;
-    }
+export default function HexView(props) {
+    const [originalContent, setOriginalContent] = useState(null);
+    const [hexlifiedContent, setHexlifiedContent] = useState(null);
+    let editorRef = React.createRef()
+    let numberRenderer = new HexViewNumberRenderer();
 
-    componentDidUpdate(prevProps) {
-        let editor = this.editor.current.editor;
-        if (this.props.mode !== prevProps.mode) {
-            if (this.props.mode === "raw") this.numberRenderer.detach(editor);
-            else this.numberRenderer.attach(editor);
+    useEffect(() => {
+        if (editorRef.current !== null) {
+            let editor = editorRef.current.editor;
+            editor.gotoLine(1);
+            if (props.mode === "hex") numberRenderer.attach(editor);
         }
-    }
+    }, []);
 
-    componentDidMount() {
-        let editor = this.editor.current.editor;
-        editor.gotoLine(1);
-        if (this.props.mode === "hex") this.numberRenderer.attach(editor);
-    }
+    useEffect(() => {
+        if (editorRef.current !== null) {
+            let editor = editorRef.current.editor;
+            if (props.mode === "hex") numberRenderer.detach(editor);
+            else numberRenderer.attach(editor);
+        }
+    });
```

Comment:
That one is tricky and I would go with something else.

useRef is hacky because it's mutable and its change doesn't trigger re-render. That's why it can't be easily used directly with useEffect (https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780).

In the same time, `ref` is just a callback that is called when component is changed (mounted/unmounted/represented by different DOM object)

That's why I'm proposing something like this:

```javascript
    const [editor, setEditor] = useState(null);

    const setEditorRef = useCallback((node) => {
        if(node) {
            setEditor(node.editor);
            node.editor.gotoLine(1);
        } else {
            setEditor(null);
        }
    }, [])

    useEffect(() => {
        if(editor && props.mode === "hex") {
            const numberRenderer = new HexViewNumberRenderer();
            numberRenderer.attach(editor);
            return () => {
                numberRenderer.detach(editor);
            }
        }
    }, [editor, props.mode])

    //...
        return (
        <AceEditor
            ref={setEditorRef}
```
1. useCallback instead of useRef. When callback is called with AceEditor object, we store the `.editor` in state and go to the first line.
2. useEffect is triggered by changed editor or mode. If mode is 'hex', we create renderer and attach it to the editor. Renderer is detached when effect is cleaned up,